### ISSUE TITLE
[HELP] Testing the next release

### ISSUE THREAD

**Issue by jokob-sk** [jokob-sk/NetAlertX#914]
Hey 👋

I'm preparing a big release and in order not to mess up your installations I'd like a few of you to test the `netalertx-dev` build. If you could create a copy of your 

- `/config`
- `/db`

Folders and spin up a separate instance with the `netalertx-dev` container, that would be super helpful. If you encounter a bug, please log it in a separate issue, or in this thread and include as much details a s possible (error log, console log (if applicable), screenshot (if applicable)).

Thanks in advance,
J

**Comment by ingoratsdorf** [comment]
Updated this morning 8:30 to Built on: 2024-12-18 | Version: 10:55:51 - Dev.
Running in docker on Alpine Linux LXC on Proxmox.

I see increases in memory and CPU consumption and most notably, a LOT more disk IO.
![image](https://github.com/user-attachments/assets/d23c8a0e-c512-4fa1-85fb-3d264fea3c39)

I checked the log folder and the app.log is getting big, although logging is switched to "off".
`25.3M -rw-r-----    1 nginx    www-data   25.3M Dec 18 11:04 app.log`
Most logging is MQTT related.
The 25 megs file has been written 8:30am to 11:04am, making it approx 10 meg an hour.

But this alone can't be it. memory consumption and IO rate are still going up.
![image](https://github.com/user-attachments/assets/63ce37a5-4136-4609-85ee-a5d8f0ec4929)

**Comment by jokob-sk** [comment]
Thanks a lot @ingoratsdorf , I will look into this and ping you to test this again if you could. I couldn't figure out how to monitor these stats on docker on my end on a Synology host. I hope to get some time on this over the next few days.

**Comment by jokob-sk** [comment]
Hey @ingoratsdorf - had some time and hope the next build in ~15 min is better. Thanks in advance for the help.

**Comment by ingoratsdorf** [comment]
I did run the "old" dev version earlier. The new dev version from yesterday f'd up my system, so had to revert back to the old stable. Some info on the disk scenario from the last couple of days:
![image](https://github.com/user-attachments/assets/1002e507-83d2-4168-b20b-2e42f682e896)

Meanwhile I cloned the stable version into a new LXC (so all the same hardware settings) cleaned db and upgraded to latest dev right now.
Looking at both in parallel in Influx, stable and dev version in comparison:
![image](https://github.com/user-attachments/assets/db5d079c-30a4-4a6e-aee8-3074e1992e90)

It's been only running for an hour now but looks rather similar. Of course the MQTT is now competeing against each other so both will be higher than they should.

The dev version is still wring heaps of MQTT debug log even though logging is set to "none" on both machines.

Will take a day at least to be able to compare. :-)

**Comment by ingoratsdorf** [comment]
Update, 7 day graph:
![image](https://github.com/user-attachments/assets/c7481b5a-ece6-4ab1-b362-4bc6352bce40)

Congrats, @jokob-sk , good job :-) Massive reduction in write access.

Now we are zooming in to the last 12 hours when I have been testing stable and latest dev version:
![image](https://github.com/user-attachments/assets/b2840efe-c1ce-487d-b124-abd8681498fa)


This morning 7:30am, I mapped /app/api to a tmpfs in docker. I figured, there is nothing in api that needs to persistent over restarts. Correct me if I am wrong.
In total since 2 days ago, that brought the disk read write from 24Mb to 1.12 Mb. It was about 1.7Mb before I mapped to tmpfs.

New docker compose file:

```
name: netalertx

services:

  netalertx:
    container_name: netalertx
    # use the below line if you want to test the latest dev image
    image: "jokobsk/netalertx-dev:latest" 
    #image: jokobsk/netalertx:latest
    network_mode: "host"        
    restart: unless-stopped
    privileged: true
    volumes:
      - /opt/netalertx/config:/app/config
      - /opt/netalertx/db:/app/db
      - /opt/netalertx/log:/app/log
      - type: tmpfs
        target: /app/api
    environment:
      - TZ=Pacific/Auckland
      - PORT=20211
```

**Comment by jokob-sk** [comment]
@ingoratsdorf taht's a great idea. I included it in the default example of the docker-compose file docs: https://github.com/jokob-sk/NetAlertX/tree/main/dockerfiles#example-1 and also in the performance tips guide: https://github.com/jokob-sk/NetAlertX/blob/main/docs/PERFORMANCE.md

**Comment by jokob-sk** [comment]
Releasing -> Closing
